test/E2ETests/Automation.CLI/DiscoveryEventsHandler.cs

```diff
@@ -13,12 +13,12 @@ public class DiscoveryEventsHandler : ITestDiscoveryEventsHandler
     /// <summary>
     /// Gets a list of Discovered tests.
     /// </summary>
-    public IList<string> Tests { get; private set; }
+    public List<string> Tests { get; } = new List<string>();
 
-    public DiscoveryEventsHandler()
-    {
-        Tests = new List<string>();
-    }
+    /// <summary>
+    /// Gets the list of messages received from the discovery process.
+    /// </summary>
+    public List<string> Messages { get; } = new List<string>();
```

Comment:
Useful when debugging so I decided to keep it.

Reply:
List?It was IList I know that's not so different

Reply:
Ideally I would expose some immutable but as it's only for testing purposes I went with simplest solution. Regarding `IList` vs `List`, I tend to prefer to return type as specific as possible and accept as generic as possible.

Reply:
ah it's test harness it's fine.